src/main/java/obelix/ObelixFeeder.java

```diff
@@ -81,50 +77,31 @@ public void run() {
 
     public boolean feed() {
         int count = 0;
-
         ObelixQueueElement result = redisQueueManager.pop();
 
         while (result != null) {
             NeoEvent event = EventFactory.build(result.data.toString());
 
             if (event != null) {
+                LOGGER.debug("Handling event: " + event);
 
-                try (Transaction tx = graphDb.beginTx()) {
-                    LOGGER.info("Handling event: " + event);
-
-                    if(metricsCollector != null) {
-                        metricsCollector.addAccumalativeMetricValue("feeded", 1);
-                    }
-
-                    event.execute(graphDb, maxRelationships);
-
-                    // Tell Obelix to rebuild the cache for the user in this event!
-                    // This will maintain the caches for all active users
-
-                    if (!usersCacheQueue.getAll().contains(new ObelixQueueElement("user_id", event.getUser()))) {
-                        usersCacheQueue.push(new ObelixQueueElement("user_id", event.getUser()));
-                    }
-
-                    tx.success();
+                if(metricsCollector != null) {
+                    metricsCollector.addAccumalativeMetricValue("feeded", 1);
+                }
 
-                } catch (TransactionFailureException e) {
-                    LOGGER.error("TransactionFailureException, need to restart");
-                    redisQueueManager.push(result);
-                    //System.exit(0);
-                } catch (NotFoundException e) {
-                    LOGGER.error("Not found exception, pushing the element back on the queue. " + e.getMessage() + ": " + result);
-                    redisQueueManager.push(result);
-                    return true;
-                } catch (DeadlockDetectedException e) {
-                    LOGGER.error("Deadlock found exception, pushing the element back on the queue" + e.getMessage() + ": " + result);
-                    redisQueueManager.push(result);
-                    return true;
-                } catch (EntityNotFoundException e) {
-                    LOGGER.error("EntityNotFoundException, pushing the element back on the queue" + e.getMessage() + ": " + result);
+                if (event.execute(graphDb, maxRelationships)) {
```

Comment:
It returns if it fails or not, all the errors are handled where the database is accessed.
This could maybe changed to an exception?